src/util-streaming-buffer.c

```diff
@@ -127,30 +135,35 @@ static inline int InitBuffer(StreamingBuffer *sb, const StreamingBufferConfig *c
 {
     sb->region.buf = CALLOC(cfg, 1, cfg->buf_size);
     if (sb->region.buf == NULL) {
-        return -1;
+        return SC_ENOMEM;
```

Comment:
Could `CALLOC` set `sc_errno`?  I think this is what you meant by using a mix of direct returns and sc_errno?  Here I think it would make sense to have success/fail results, then `sc_errno` could be consulted.

Reply:
Moved setting to the allocators.